modules/speech-to-text-async-impl/src/main/java/org/opencastproject/speechtotext/async/impl/SpeechToTextWorkflowSchedulerQuartz.java

```diff
@@ -0,0 +1,403 @@
+      try {
+        List<SpeechToTextControl> stts = database.findByWorkflowId(wfId);
+        mpId = stts.get(0).getMediaPackageId();
+        boolean doAttach = stts.stream()
+                .allMatch(stt -> SpeechToTextControl.Status.TranscriptionDone == stt.getStatus());
+
+        if (!doAttach) {
+          logger.debug(
+                  "Will not process media package {} workflow {} this time because not all STT jobs have finished.",
+                  mpId, wfId);
+          continue;
+        }
+
+        List<JpaJob> jobs = stts.stream().map(sst -> sst.getJob()).collect(Collectors.toList());
+
+        // Start workflow to attach subtitles to that media package
+        WorkflowInstance wfInstance = startWorkflow(mpId, jobs);
+        if (wfInstance == null) {
+          logger.warn("Could not start workflow for mp {}", mpId);
+          // Will try again in the next run; there may be another workflow in progress for that media package
+          continue;
+        }
+        logger.info("Workflow to attach subtitles started for mp: {}, wf instance id: {}", mpId, wfInstance.getId());
+        // If success, update status in database
+        database.updateStatusByJob(SpeechToTextControl.Status.WorkflowInProgress, jobs.toArray(new JpaJob[0]));
+      } catch (Exception e) {
+        logger.warn("Error when starting workflow to attach subtitles to media package: {}", mpId, e);
+      }
+    }
+  }
+
+  @Override
+  public String getScannerName() {
+    return SCANNER_NAME;
+  }
+
+  /**
+   * Starts a workflow on the latest version of the media package.
+   *
+   * @param mpId
+   *          The media package id
+   * @param wfDefId
+   *          The workflow definition id
+   * @return Workflow instance started
+   */
+  WorkflowInstance startWorkflow(String mpId, List<JpaJob> jobs) {
+    // Is there a snapshot?
+    if (!assetManager.snapshotExists(mpId)) {
+      // Media package not archived yet.
+      logger.warn("Media package {} has not been archived yet.", mpId);
+      return null;
+    }
+
+    try {
+      WorkflowDefinition wfDef = workflowService.getWorkflowDefinitionById(attachWorkflowDef);
+      Workflows workflows = wfUtil != null ? wfUtil : new Workflows(assetManager, workflowService);
+      Set<String> mpIds = Collections.singleton(mpId);
+      Map<String, String> wfConfig = new HashMap<String, String>();
+      wfConfig.put(JOBS_WORKFLOW_CONFIGURATION,
+              jobs.stream().map(j -> String.valueOf(j.getId())).collect(Collectors.joining(",")));
+      List<WorkflowInstance> wfList = workflows
+              .applyWorkflowToLatestVersion(mpIds, ConfiguredWorkflow.workflow(wfDef, wfConfig));
+      WorkflowInstance wf = wfList.size() > 0 ? wfList.get(0) : null;
+      logger.info("Workflow {} {}started for mp {}{}", attachWorkflowDef, wf != null ? "" : "NOT ", mpId,
+              wf != null ? ": " + wf.getId() : "");
```

Comment:
While this is more verbose, it makes for *way* easier reading
```suggestion
      if (wf != null) {
        logger.info("Workflow {} started for mp {}: {}", attachWorkflowDef, mpId, wf.getId());
      } else {
        logger.info("Workflow {} NOT started for mp {}", attachWorkflowDef, mpId);
      }
```

Reply:
Fixed.